augur/merge.py

```diff
@@ -354,6 +379,57 @@ def merge_metadata(args):
             print_info(f"WARNING: Skipped deletion of {db_path} due to error, but you may want to clean it up yourself (e.g. if it's large).")
 
 
+def merge_sequences(args):
+    if not args.skip_input_sequences_validation:
+        for s in args.sequences:
+            print_info(f"Validating {s!r}…")
+
+            with NamedTemporaryFile() as dup_num_file:
+                command = f"""
+                    {augur} read-file {shquote(s)} |
+                    {seqkit()} rmdup --dup-num-file {shquote(dup_num_file.name)}
+                """
+
+                returncode = run_command(command, print_stdout=False)
+
+                if returncode != 0:
+                    raise AugurError(f"Validation failed for {s!r}.")
+
+                dup_num_file.seek(0)
+                
+                if duplicates := list(parse_seqkit_dup_num_file(dup_num_file.read().decode().strip())):
+                    raise AugurError(f"File {s} has duplicate IDs: {', '.join(repr(d) for d in duplicates)}")
+
+    print_info(f"Merging sequences and writing to {args.output_sequences!r}…")
+
+    # Reversed because seqkit rmdup keeps the first entry but this command
+    # should keep the last entry.
+    # "echo;" adds a newline character to support FASTA files that are missing one at the end.
+    # This is fine because extraneous newline characters are stripped by seqkit.
+    command = f"""
+        (for f in {" ".join(shquote(s) for s in reversed(args.sequences))}; do
+            {augur} read-file "$f";
+            echo;
+        done) |
```

Comment:
_non-blocking_

This pays Augur startup cost N times, and is also more to digest code-wise. I'd be inclined to make `augur read-file` able to take more than one file and put the loop there in Python.

Reply:
Good idea, will look into this.

Reply:
Created a separate PR for this: #1815

Using it here:

https://github.com/nextstrain/augur/blob/1979c61036850e5de021908797b7f15be06267d0/augur/merge.py#L399-L403